src/main/java/com/teksystems/TekcampExerciseSpring/Exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,31 @@
+package com.teksystems.TekcampExerciseSpring.Exception;
+
+import java.util.Date;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+	
+	@ExceptionHandler(ResourceNotFoundException.class)
+	public ResponseEntity<?> handleResourceNotFoundException(ResourceNotFoundException exception, WebRequest request){
+		ErrorDetails errorDetails = new ErrorDetails(new Date(), exception.getMessage(), request.getDescription(false));
+		return new ResponseEntity(errorDetails, HttpStatus.NOT_FOUND);
+	};
+	@ExceptionHandler(ApiException.class)
+	public ResponseEntity<?> handleApiException(ApiException exception, WebRequest request){
+		ErrorDetails errorDetails = new ErrorDetails(new Date(), exception.getMessage(), request.getDescription(false));
+		return new ResponseEntity(errorDetails, HttpStatus.NOT_FOUND);
+	};
+	
+	@ExceptionHandler(Exception.class)
+	public ResponseEntity<?> handleGlobalException(Exception exception, WebRequest request){
+		ErrorDetails errorDetails = new ErrorDetails(new Date(), exception.getMessage(), request.getDescription(false));
+		return new ResponseEntity(errorDetails, HttpStatus.NOT_FOUND);
+	};
```

Comment:
you have 3 different handlers that are all handled the exact same way, that means it's duplicate code and unnecessary.